docarray/array/storage/weaviate/backend.py

```diff
@@ -174,21 +185,17 @@ def _load_or_create_weaviate_schema(self):
 
     def _update_offset2ids_meta(self):
         """Update the offset2ids in weaviate the the current local version"""
-        if self._offset2ids_wid is not None and self._client.data_object.exists(
-            self._offset2ids_wid
-        ):
-            self._client.data_object.update(
-                data_object={'_offset2ids': self._offset2ids},
-                class_name=self._meta_name,
-                uuid=self._offset2ids_wid,
-            )
-        else:
+        if self._offset2ids_wid is None:
             self._offset2ids_wid = str(uuid.uuid1())
-            self._client.data_object.create(
-                data_object={'_offset2ids': self._offset2ids},
-                class_name=self._meta_name,
-                uuid=self._offset2ids_wid,
-            )
+            
+        # this call will either create or replace an object (if uuid exists)
+        # since Meta class has a single property: '_offset2ids', we can replace
+        # the object instead of updating it.
+        self._client.batch.add_data_object(
+            data_object={'_offset2ids': self._offset2ids},
+            class_name=self._meta_name,
+            uuid=self._offset2ids_wid,
+        )
```

Comment:
Is it a good idea to use context manager `with self._client.batch` instead, so as to automatically flushing batch?

Reply:
One would use the context manager when importing/creating data in the same function, otherwise it will flush the objects once it exits the context manager. In this case it would, if the batch is empty and we call this method it will flush the batch with only one object. This is not something we want, because it is the same as creating the object explicitly.